checker/jtreg/index/UnneededSuppressionsTest.java

```diff
@@ -36,4 +36,12 @@ void method5() {}
 
     @SuppressWarnings("purity")
     void method6() {}
+
+    // TODO: It is a bug that this is not reported as an unused suppression
+    @SuppressWarnings("index:foo.bar.baz")
```

Comment:
Because `foo.bar.baz` is not a message key for the Index Checker, this is not reported as unused.

Reply:
1. This behavior is not documented.
2. Do you believe this is desirable behavior?  It means that a user who typos a message gets no warning about that typo.

Reply:
1. This behavior was documented, but was removed 1abced10d7df1e80a7909fed92b6900739cb1c03.  I suggest we add back the line `These warnings are only issued if the \<@SuppressWarnings>
  contains a key that would suppress any warning from this checker.`  in the introduction.tex. 

2. The purpose of this option is to find warning suppressions that at one time did suppress a checker warning.  So at the time the suppression is written a checker is issuing a warning.  If a user tries to suppress that warning, but typos the suppression string, then the checker will still issue a warning.  And thus that warning tells the user there's a typo.